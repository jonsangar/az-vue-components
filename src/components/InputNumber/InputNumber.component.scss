$input-size: 30px;

.input-number{
    display: flex;
    align-items: flex-end;
    justify-content: center;
    margin: 0 quarter();

    &__title {
      width: $input-size;
      height: $input-size;
      margin-right: quarter();
    }

    &__value{
      min-width: $input-size;
      height: $input-size;
      color: color(gray3);
      text-align: center;
      background-color: transparent;
      border: none;
    }

    &__button{
      width: $input-size;
      height: $input-size;
      text-align: center;
      background-color: color(gray8);
      border-radius: 5px;
    }
}
