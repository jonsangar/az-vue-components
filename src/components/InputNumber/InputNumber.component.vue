<template lang="pug">
.input-number
  .input-number__title(
    v-if="title")
    | {{ title }}
  .input-number__button
    az-icon(
      name="less"
      scale="0.8"
      @click="decrement")
  .input-number__value
    | {{ localValue }} {{ sufix ? sufix : '' }}
  .input-number__button
    az-icon(
      name="more"
      scale="0.8"
      @click="increment")
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      required: false
    },

    value: {
      type: Number,
      required: true
    },

    min: {
      type: Number,
      required:false
    },

    max: {
      type: Number,
      required: false
    },

    step: {
      type: Number,
      default: 1
    },

    valueList: {
      type: Array,
      required: false
    },

    sufix: {
      type: String,
      required: false
    }
  },

  data() {

    return {
      localValue: this.value
    }

  },

  watch: {
    localValue(newValue) {
      this.$emit('input', newValue);
    },

    value(newValue) {
      this.localValue = newValue;
    }
  },

  methods: {
    increment() {
      if( this.valueList && this.valueList.length > 0){
        const index = this.valueList.indexOf(this.localValue);
        const hasNext = (index + 1) < this.valueList.length ;
        if (!hasNext) return;
        this.localValue = this.valueList[index + 1];
      }

      this.$emit('click', this.localValue);
    },

    decrement() {

      if( this.valueList && this.valueList.length > 0){
        const index = this.valueList.indexOf(this.localValue);
        const hasPrev = (index - 1) >= 0;
        if (!hasPrev) return;
        this.localValue = this.valueList[index - 1];
      }

      this.$emit('click', this.localValue);

    }
  }
}
</script>

<style lang="scss" src="./InputNumber.component.scss" scoped></style>
