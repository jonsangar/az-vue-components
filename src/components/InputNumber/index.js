import InputNumber from './InputNumber.component.vue';

export default InputNumber;
