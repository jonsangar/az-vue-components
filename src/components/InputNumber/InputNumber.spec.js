import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerÃ­as de estilos e iconos
import SvgIcon from 'vue-svgicon'
import InputNumber from './InputNumber.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('InputNumber.vue', () => {
  const build = () => {
    const wrapper = shallowMount(InputNumber, {
      localVue,
      propsData: {
        value: 1,
        valueList: [0,1,2,3,4,5]
      }
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente InputNumber con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()

  })

  it('Comprueba que emite el evento click al incrementar', async () => {
    const { wrapper } = build()

    await wrapper.vm.increment()
    expect(wrapper.emitted().click).toBeTruthy()
    expect(wrapper.emitted().click.length).toBe(1)
    expect(wrapper.emitted().click[0]).toEqual([2,])
  });

  it('Comprueba que emite el evento click al decrementar', async () => {
    const { wrapper } = build()

    await wrapper.vm.decrement()
    expect(wrapper.emitted().click).toBeTruthy()
    expect(wrapper.emitted().click.length).toBe(1)
    expect(wrapper.emitted().click[0]).toEqual([0,])
  });
})
