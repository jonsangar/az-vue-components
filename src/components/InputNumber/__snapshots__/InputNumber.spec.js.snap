// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber.vue Comprueba renderizado del componente InputNumber con snapshot 1`] = `
<div class="input-number">
  <!---->
  <div class="input-number__button">
    <az-icon-stub name="less" width="" height="" scale="0.8" fill="true"></az-icon-stub>
  </div>
  <div class="input-number__value">1 </div>
  <div class="input-number__button">
    <az-icon-stub name="more" width="" height="" scale="0.8" fill="true"></az-icon-stub>
  </div>
</div>
`;
